translator/GCP_cloud_function/cloud_function/main.py

```diff
@@ -42,28 +41,59 @@ def translate_newest_icone_to_wzdx(event, context):
   return
 
 def get_ftp_file(url) :
-  with closing(request.urlopen(url)) as r:
-    return r.read().decode('utf-8-sig')
-
+  return urllib.request.urlopen(url).read().decode('utf-8-sig')
 
 def parse_xml(xml_string):
   return xmltodict.parse(xml_string)
 
 def get_wzdx_schema(schema_file_name):
-  return json.loads(open(schema_file_name,'r').read())
+  try:
+    schema_string=open(schema_file_name,'r').read()
+  except FileNotFoundError as e:
+    raise RuntimeError('invalid schema: file does not exist')
+  if not schema_string:
+    raise RuntimeError('invalid schema: empty schema')
+  try:
+    schema_obj= json.loads(schema_string)
+    jsonschema.validate('', schema_obj)  
+  except json.decoder.JSONDecodeError as e :
+    raise RuntimeError('invalid schema: not valid json') from e
+  except jsonschema.SchemaError as e :
+    raise RecursionError('invalid schema: schema failed validation') from e
```

Comment:
Was this meant to be a RuntimeError? RecursionError is "raised when the interpreter detects that the maximum recursion depth is exceeded". This isn't using recursion so we shoudln't be throwing this error.

Reply:
I changed it to RuntimeError